### PR TITLE
Implement theming `PapercupStyle()`, Flutter 3.0 support and more

### PR DESCRIPTION
**Pull request by CharlesMangwa**
Hi @aguilaair!

This PR includes a couple of changes listed as follow:

## Flutter 3 support

The change made in 1aa632234a7163095bacf502537427348fdfa93a fixes the redundant null-aware operators warning introduced in Flutter 3.0, while keeping [backwards compatibility](https://docs.flutter.dev/development/tools/sdk/release-notes/release-notes-3.0.0#your-code) for previous Flutter versions.

## Full theming support via new `PapercupsStyle`

Even with the work made for #18, I encountered some visual discrepancies after enabling dark mode in my app via:

```dart
MaterialApp(
  theme: Themes.lightMode, // Light mode ThemeData
  darkTheme: Themes.darkMode, // Dark mode ThemeData
)
```

That's why I exposed most of the styling params via the new `PapercupsStyle` class. The new API has been detailed in the updated README. Please feel free to double-check I didn't make any mistake there 😅.

#### ⚠️ BREAKING CHANGE

This change also required a breaking change to `PaperCupsWidget` & `Props` in order to collocate all visual params.

## Prefix all user-facing classes with `Papercups`

In order to add consistency and avoid potential naming conflicts, I add `Papercups` as a prefix to all the user-facing classes (`PapercupsCustomerMetadata`, `PapercupsProps`, `PapercupsStyle` & `PapercupsIntl`).

#### ⚠️ BREAKING CHANGE

In the same spirit, I made another breaking change which was renaming `PaperCupsWidget` to `PapercupsWidget`, so that it'd follow the official spelling.

## Update README with missing params

On top of the new `PapercupsStyle` and moving some of the old `PapercupsWidget` & `PapercupsProps` params around, I also added some existing params that were not documented yet.

---

I hope all these changes make sense to you and can fit in with what you had planned for the upcoming v3.

Let me know if you'd like me to make some changes! 🙌

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,57 +31,382 @@ That should get you up and running in just a few seconds ⚡️.
...
-| Parameters | Type | Value | Default |
-| :--- | :--- | :----- | :------ |
-| **`email`** | `String` | The customer's email| N/A |
-| **`externalId`** | `String` | The customer's external ID | N/A |
-| **`name`** | `String` | The customer's name | N/A |
-| **`otherMetadata`** | `Map<String, String>` | Extra metadata to pass such as OS info | N/A |
+### Available `PapercupsWidget` arguments
+| Parameter           | Type      | Value                                                                                                                                    | Default   |
+| :------------------ | :-------- | :--------------------------------------------------------------------------------------------------------------------------------------- | :-------- |
+| **`props`**         | `Props`   | **Required**. This is where all of the config for the chat is contained.                                                                 | N/A       |
```
Comment (aguilaair) [inline README.md:37]: PapercupsProps

File: lib/papercups_flutter.dart [hunk 2]
```diff
@@ -63,19 +51,17 @@ class _PaperCupsWidgetState extends State<PaperCupsWidget> {
 
   @override
   void dispose() {
-    if (_channel != null) _channel!.close();
-    if (_conversationChannel != null) _conversationChannel!.close();
-    if (_socket != null) _socket!.dispose();
+    _channel?.close();
+    _conversationChannel?.close();
+    _socket?.dispose();
```
Comment (aguilaair) [inline lib/papercups_flutter.dart:56]: Thanks, migrating from non-nullsafe versions of Dart XD

File: lib/papercups_flutter.dart [hunk 3]
```diff
@@ -155,20 +135,14 @@ class _PaperCupsWidgetState extends State<PaperCupsWidget> {
...
     super.didUpdateWidget(oldWidget);
-    if (oldWidget.props.primaryColor != widget.props.primaryColor) {
-      if ((widget.props.primaryColor != null &&
-              widget.props.primaryColor!.computeLuminance() > 0.5) ||
-          (widget.props.primaryGradient != null &&
-              widget.props.primaryGradient!.colors[0].computeLuminance() >
-                  0.5) ||
-          (widget.props.primaryColor == null &&
-              Theme.of(context).primaryColor.computeLuminance() > 0.5)) {
+    if (oldWidget.props.style.primaryColor != widget.props.style.primaryColor) {
```
Comment (aguilaair) [inline lib/papercups_flutter.dart:140]: Maybe we want to optionally disable computeLuminance if we're already setting everything else up in `PapercupsStyle`?

File: lib/papercups_flutter.dart [hunk 4]
```diff
@@ -190,15 +164,19 @@ class _PaperCupsWidgetState extends State<PaperCupsWidget> {
     _conversationChannel = c;
   }
 
+  /// This allows a value of type T or T?
+  /// to be treated as a value of type T?.
+  ///
+  /// We use this so that APIs that have become
+  /// non-nullable can still be used with `!` and `?`
+  /// to support older versions of the API as well.
+  T? _ambiguate<T>(T? value) => value;
```
Comment (aguilaair) [inline lib/papercups_flutter.dart:173]: Nice!

**Review by aguilaair — COMMENTED** [review]
Just a comment regarding the README, but other than that looks great!

File: lib/papercups_flutter.dart (hunk 3 above)
Comment (CharlesMangwa) [reply lib/papercups_flutter.dart:140]: I think it'd make sense to keep it if some people set `primaryColor`/`primaryGradient`, but not `userBubbleTextStyle `/`userAttachmentTextStyle`. By keeping the `computeLuminance` the latter 2 - which use `widget.textColor` -  could then have their luminance be automatically adapted to the color the user provided.

**Comment by CharlesMangwa** [comment]
Also added some more refactoring for non-nullsafe snippets here and there.

**Comment by aguilaair** [comment]
I'll do my best to release 3.0 this weekend, thank you so much for all this @CharlesMangwa

**Comment by CharlesMangwa** [comment]
Awesome, thanks for the super fast merge on again! 🙌  And thanks in advance for the v3 but don't worry, no hurry!

**Pull request merged**